include/LightGBM/utils/common.h

```diff
@@ -336,6 +336,22 @@ static inline int64_t Pow2RoundUp(int64_t x) {
   return 0;
 }
 
+inline void Softmax(std::vector<float>* p_rec) {
+  std::vector<float> &rec = *p_rec;
+  float wmax = rec[0];
```

Comment:
@xuehui1991 to be honest, I don't know. I copied it from [xgboost](https://github.com/dmlc/xgboost/blob/master/src/common/math.h#L30). The result is same with or without it.